api/envoy/config/core/v3/health_check.proto

```diff
@@ -186,6 +186,19 @@ message HealthCheck {
     // the :ref:`hostname <envoy_v3_api_field_config.endpoint.v3.Endpoint.HealthCheckConfig.hostname>` field.
     string authority = 2
         [(validate.rules).string = {well_known_regex: HTTP_HEADER_VALUE strict: false}];
+
+    // Specifies a list of HTTP headers that should be added to each request that is sent to the
+    // health checked cluster. For more information, including details on header value syntax, see
+    // the documentation on :ref:`custom request headers
+    // <config_http_conn_man_headers_custom_request_headers>`.
+    repeated HeaderValueOption request_headers_to_add = 3
+        [(validate.rules).repeated = {max_items: 1000}];
+
+    // Specifies a list of HTTP headers that should be removed from each request that is sent to the
+    // health checked cluster.
+    repeated string request_headers_to_remove = 4 [(validate.rules).repeated = {
```

Comment:
What is the use case for removing headers? I'm not sure I understand it in the HTTP case either. Is this for removing some headers generated by Envoy during the HC? CC @mattklein123

Reply:
I agree it's not very useful, actually the only metadata fields I found for the GRPC default health checker in my test were: ["authority", "content-type", "user-agent"], it might be a little bit more useful for the http health checker for which I guess there're more headers that can be removed : ), I'm totally fine if we only want to keep the request_headers_to_add field for GRPC health checker.

Reply:
For HTTP I don't recall why it was added, but I could see a bunch of reasons why people would want it, for example obscuring some data from backend responses.

For gRPC it might be less useful. I would probably only add it if you need it. We can always add it later?

Reply:
ack, removed the "request_headers_to_remove" option and renamed "request_headers_to_add" to "initial_metadata" to align with GRPC terms as htuch suggested.